lib/portlayer/network/context.go

```diff
@@ -942,14 +942,17 @@ func (c *Context) RemoveIDFromScopes(op trace.Operation, id string) ([]*Endpoint
 		// save the endpoint info
 		e := con.Endpoint(s).copy()
 
-		if err := s.RemoveContainer(con); err != nil {
-			return nil, err
-		}
-
 		aliases = append(aliases, c.populateAliases(con, s, e)...)
 		endpoints = append(endpoints, e)
 	}
 
+	// remove the container from all bound scopes
+	for _, ne := range endpoints {
+		if err := ne.scope.RemoveContainer(con); err != nil {
```

Comment:
I guess it would be good to add a comment here that ne.scope.RemoveContainer modified con.endpoints